release/authorization/README.md

```diff
@@ -0,0 +1,407 @@
+# gNMI Authorization Protocol
+
+## Objective
+
+This proto definition and the reference code(to be delivered seperately) serve
+to describe an authorization framework for controlling which OpenConfig paths of
+a network device users can access using a gNMI micro-serivice. The authorization
+policy is initially intended to be deployed to a device, with the ability to
+define:
+
+*   Policy rules - each rule defines a single authorization policy.
+*   Groups of users - as a method to logically group users in the administrative
+    domain, for instance: operators or administrators.
+*   Users which are individual users as part of either rules or in groups.
+    Authentication information is not included in this Authorization
+    configuration.
+
+Policy rules are matched based on the best match for the authorization request,
+not the first match against a policy rule. Best match enables a configuration
+which permits a user or group access to particular OpenConfig paths while
+denying subordinate portions of the permitted paths, or the converse.
+
+Best, or most specific, match is that which has the longest match to the
```

Comment:
Let's add a section on what exactly is "best match" means.
Some suggestion text (just for reference):

- Authorization will build a tree with the configuration. If there is conflict in the config (with same path, same user, same level, but different action), then ...
- Authorization will take a user name, gnmi path, access level (read or write), and return a boolean (or match rule).
- Authorization will walk the tree with the given gnmi path. Record the matched rule. Final decision will be applied on the last matched rule.
- Matched rule means that the user must match the user name or group in the config node, and/or the access level must match in the config node.
- In key check, authorization will first check the specific key match subtree. If there is a match, return (instead of checking the wildcard subtree) EVEN if the wildcard subtree might return a longer matched rule. If there is no match, continue with the wildcard subtree.
- For multiple key check (just suggestion), authorization will sort the keys. Apply specific match with the keys in order.
- ...

Reply:
@mint570 I like your additional text options. I added text and arranged the README to take it into account for most of your suggested bits. I shied away from the text about implementation details for 'checking specific keys then wildcards', and for sorting multiple key checks because they both sounds very much like implementation specific optimizations.

I hope the rewording brings the clarity you were after.
Happy to clarify more if required.